m0t1fs/linux_kernel/file.c

```diff
@@ -2317,70 +2317,80 @@ static int pargrp_iomap_paritybufs_alloc(struct pargrp_iomap *map)
 			   "data_buf.", map->pi_ioreq);
 }
 
-static int pargrp_iomap_populate(struct pargrp_iomap        *map,
-				 struct m0_ivec_varr_cursor *cursor)
+/**
+ * Set segment by countiguous user data from cursor until the
+ * parity group end.
+ */
+static m0_bindex_t seg_set(struct pargrp_iomap *map, uint32_t seg,
+			   struct m0_ivec_varr_cursor *cur, m0_bindex_t grpend)
+{
+	m0_bindex_t end = m0_ivec_varr_cursor_conti(cur, grpend);
+
+	V_INDEX(&map->pi_ivv, seg) = m0_ivec_varr_cursor_index(cur);
+	V_COUNT(&map->pi_ivv, seg) = end - V_INDEX(&map->pi_ivv, seg);
+
+	return end;
+}
+
+/* Make segment's index sz-aligned by rounding it up. */
+static void seg_round_up_idx(struct pargrp_iomap *map, uint32_t seg,
+			     uint64_t sz)
+{
+	m0_bindex_t idx = m0_round_up(V_INDEX(&map->pi_ivv, seg) + 1, sz);
```

Comment:
I need to understand why we need to increase the index by one,
V_INDEX(&map->pi_ivv, seg) + 1

Maybe some explanation is helpful.

Reply:
Because we want to round it up. Say, if index is 0 - it won't be rounted up, will it? But if we increase it by 1 - it will be rounded.